src/algorithms/cache.rs

```diff
@@ -0,0 +1,280 @@
+use crate::{enumerate_mask, Correctness, Guess, Guesser, DICTIONARY, MAX_MASK_ENUM};
+use once_cell::sync::OnceCell;
+use std::borrow::Cow;
+
+static INITIAL: OnceCell<Vec<(&'static str, f64, usize)>> = OnceCell::new();
+static PATTERNS: OnceCell<Vec<[Correctness; 5]>> = OnceCell::new();
+
+// Attempted to use vec of vecs for compute cache, but was much slower
+static COMPUTES: OnceCell<(usize, Vec<OnceCell<u8>>)> = OnceCell::new();
```

Comment:
To check my own understanding here — this is a `Vec`, not an array, because the number of words isn't known at compile-time? I think we should rectify that by introducing a `build.rs` that parses `dictionary.txt` at build time and emits a Rust array directly into a source file that we can then include. Will make this a little nicer.

Reply:
To take this even further, I wonder if we should make this a `thread_local!` holding `Cell<u8>`. That way it'll be way cheaper to access in the single-threaded case.